frame/contracts/src/storage.rs

```diff
@@ -201,6 +216,65 @@ impl<T: Config> ContractInfo<T> {
 		})
 	}
 
+	/// Sets and returns the contract base deposit.
+	/// `upload_deposit` is the balance of the deposit paid for uploading the contract.
+	pub fn update_base_deposit(&mut self, upload_deposit: BalanceOf<T>) -> BalanceOf<T> {
```

Comment:
IMHO it would be better to pass in a reference to the `CodeInfo` struct here. Passing around numbers/primitives is always a bit dangerous.